java/com/currencyone/example/Example.java

```diff
@@ -57,13 +63,17 @@ public static String makeSignedRequest(String method, String uri) throws Excepti
     }
 
     public static String connectionResponseToString(HttpURLConnection connection) throws IOException {
-        InputStream is;
-        if (200 <= connection.getResponseCode() && connection.getResponseCode() <= 299) {
-            is = connection.getInputStream(); 
-        } else {
-            is = connection.getErrorStream();
+        try (InputStream is = connection.getInputStream()) {
+            return inputStreamToString(is); 
+        } catch (IOException e) {
+            try (InputStream is = connection.getErrorStream()) {
+                throw new IllegalArgumentException(inputStreamToString(is));
```

Comment:
`IllegalArgumentException`? Why?

Reply:
- connection was lost: `IllegalArgumentException` :thinking: 
- server error: `IllegalArgumentException` :thinking: 
- API arguments were illegal: no exception… :thinking:

Cannot think of more misleading exception than the very `IllegalArgumentException` in this line. `RuntimeException` would be just fine, it's an example anyway.